lib/aws-xray-sdk/model/entity.rb

```diff
@@ -143,7 +143,9 @@ def to_h
       else
         h[:in_progress] = true
       end
-      h[:subsegments] = subsegments unless subsegments.empty?
+
+      h[:subsegments] = subsegments.map(&:to_h) unless subsegments.empty?
```

Comment:
Oj is used for MRI and an object can be serialized as long as `to_h` is implemented for its members. What is the intention of explicitly calling `to_h` here?

Reply:
In our dev environment, `entity.to_json` have raised error when emitter sent payload, object did't do to_h as expected

Reply:
Is it similar to https://github.com/aws/aws-xray-sdk-ruby/issues/15? I'm still working on solving the root problem. Is this a workaround that works for you? I currently prefer not to merge this particular change before the root problem is solved as it masks performance impact from improper serializer.

Reply:
Yes, but i don't think it is a good idea to change the encoder of the rails, we need to evaluate if the replacing will bring some side effects, for commercial using stable is very important. And also i don't think it is a good idea for the Tracing SDK to affect the underlying of application.

Reply:
Yes, my traceback is the same as #15  . Because `multi_json` don't pick oj as json encoder here, so convert object to hash manually(`subsegments.map(&:to_h)` and `@data.keys.each_with_object({}) do |key, h|`) works. Maybe you can take this as temporary patch before the root program solved

Reply:
I'm ok with merging this for corner cases where `Oj` is not correctly picked up. Ideally users should never worry about the serialization details and the SDK should simply work with its maximum performance. But for cases like you encountered, extra tuning might be necessary and please be aware of the extra overhead of active support serializer. I will create a separate issue to track this once this PR is released so SDK users are aware of the potential issue.